firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/RemoteConfigDeferredProxy.java

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.crashlytics.internal;
+
+import com.google.firebase.crashlytics.internal.metadata.UserMetadata;
+import com.google.firebase.inject.Deferred;
+import com.google.firebase.remoteconfig.interop.FirebaseRemoteConfigInterop;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class RemoteConfigDeferredProxy {
+  private final Deferred<FirebaseRemoteConfigInterop> remoteConfigInteropDeferred;
+  private final AtomicReference<UserMetadata> userMetadata = new AtomicReference<>(null);
+
+  public RemoteConfigDeferredProxy(
+      Deferred<FirebaseRemoteConfigInterop> remoteConfigInteropDeferred) {
+    this.remoteConfigInteropDeferred = remoteConfigInteropDeferred;
+  }
+
+  public void setupListener(UserMetadata metadata) {
+    userMetadata.set(metadata);
+    if (userMetadata.get() == null) {
+      Logger.getLogger().w("Didn't successfully register with UserMetadata for rollouts listener");
+      return;
+    }
+    CrashlyticsRemoteConfigListener listener =
+        new CrashlyticsRemoteConfigListener(userMetadata.get());
+    remoteConfigInteropDeferred.whenAvailable(
+        remoteConfigInteropProvider -> {
+          FirebaseRemoteConfigInterop interop = remoteConfigInteropProvider.get();
+          interop.registerRolloutsStateSubscriber("crashlytics", listener);
```

Comment:
`registerRolloutsStateSubscriber` should just take the listener as a parameter

Reply:
Seems like need to touch interop, shall we address it in a followup pr?

Reply:
Seems I have to add the namespace:
![Screenshot 2023-08-30 at 10 47 33 PM](https://github.com/firebase/firebase-android-sdk/assets/16548721/4112aaf0-ac33-40a8-bec2-c25d591442f3)